function loadImg(src) {
  const promise = new Promise((resolve, reject) => {
    const img = document.createElement('img')
    img.onload = () => {
      resolve(img)
    }
    img.onerror = () => {
      reject(new Error(`图片加载失败 ${src}`))
    }
    img.src = src
  })
  return promise
}

async function loadImg1() {
  const src1 = 'http://www.imooc.com/static/img/index/logo_new.png'
  const img1 = await loadImg(src1)
  return img1
}

async function loadImg2() {
  const src2 = 'https://avatars3.githubusercontent.com/u/9583120'
  const img2 = await loadImg(src2)
  return img2
}

(async function () {
  // 注意：await 必须放在 async 函数中，否则会报错
  try {
    // 加载第一张图片
    const img1 = await loadImg1()
    console.log(img1, img1.width, img1.height)
    // 加载第二张图片
    const img2 = await loadImg2()
    console.log(img2, img2.width, img2.height)
  } catch (ex) {
    console.error(ex)
  }
})()